### ISSUE TITLE
Plugin is getting activated even if woocommerce is not present

### ISSUE THREAD

**Issue by blockonomics** [blockonomics/woocommerce-plugin#107]
- This confuses users as he is searching for blockonomics in settings
- See how other plugins handle this. I think some don't even install without dependency

**Comment by blockonomics** [comment]
The good side effect of this is that Pay with Bitcoin started working when WC was installed later ! But not sure if this behaviour is really desired

**Comment by shivaenigma** [comment]
Please state proposed solution before coding this

**Comment by jusasiiv** [comment]
This was already done in commit https://github.com/blockonomics/woocommerce-plugin/commit/b0fcdb4555451700898b348cf227a87e3e940892

Solution is based on these two stack overflow answers 
https://wordpress.stackexchange.com/questions/251114/stop-a-plugin-in-the-activation-process-when-a-certain-wp-version-is-not-met-the 
and https://wordpress.stackexchange.com/questions/189208/check-for-dependent-plugin-and-if-false-dont-activate-plugin

Basically we stop the activation process with a die(). This displays an error message using wp's own activation and error libraries. Also supports if users are using wp-cli or other installation methods

**Comment by shivaenigma** [comment]
I am concered about removal of
`if (is_plugin_active('woocommerce/woocommerce.php') || class_exists('WooCommerce')) {` from the main code.  Lets cross check that is not not populating Settings Menu / Other Hooks

https://wordpress.stackexchange.com/questions/189208/check-for-dependent-plugin-and-if-false-dont-activate-plugin looks a better than `die()`. Lets use that

**Comment by jusasiiv** [comment]
> I am concered about removal of
`if (is_plugin_active('woocommerce/woocommerce.php') || class_exists('WooCommerce')) {` from the main code. Lets cross check that is not not populating Settings Menu / Other Hooks

There is a check for WooCommerce in plugin init https://github.com/blockonomics/woocommerce-plugin/blob/version-1.7.2/blockonomics-woocommerce.php#L51 so if WooCommerce is not active nothing will init
